### PR TITLE
fix: device pixel ratio exposing rendering artifacts

### PR DESCRIPTION
**Pull request by christjt**
# Description

Fixes rendering issue caused by users setting device pixel ratio. For more context see: #2205 

Jira issue: https://cognitedata.atlassian.net/browse/REV-381 

This fix will also prevent users from circumventing our resolution cap by increasing the device pixel ratio. Ticket for letting users control this cap is added here: https://cognitedata.atlassian.net/browse/REV-382

# Checklist:

Here is a checklist that should completed before merging this given feature. 
Any shortcomings from the items below should be explained and detailed within the contents of this PR.

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
Not applicable since we are now compliant with normal threejs convention and are covered by: https://threejs.org/manual/#en/responsive
- [x] I have added unit and visuals tests to prove that my feature works to the best of my ability.
I added a sanity test in a existing visual test, but unit testing the resizeIfNeccesary is close to impossible with current architecture. Have added ticket for this: https://cognitedata.atlassian.net/browse/REV-385
- [x] I have refactored the code for readability to the best of my ability.
- [x] I have checked that my changes do not introduce regressions in the public documentation.
- [x] I have outlined any known defects / lacking cabilities in the contents of this PR.
- [x] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [x] I have added TSDoc to any public facing changes.
- [x] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [x] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [x] I have asked peers to test this feature.

### CODE REVIEW

File: examples/src/pages/Migration.tsx [hunk 1]
```diff
@@ -360,6 +360,8 @@ export function Migration() {
 
       const inspectNodeUi = new InspectNodeUI(gui.addFolder('Last clicked node'), client, viewer);
 
+      viewer.renderer.setPixelRatio(window.devicePixelRatio);
```
Comment (pramodcog) [inline examples/src/pages/Migration.tsx:363]: you are setting pixel ratio at https://github.com/cognitedata/reveal/blob/2644eb4d78284cda71d6fcab8e56abf28c692cff/viewer/packages/rendering/app/index.ts#L61 why do we need to set them here

**Review by pramodcog — CHANGES_REQUESTED** [review]

**Comment by pramodcog** [comment]
@christjt Does `potree` library take care of the pixel ratio?

**Comment by christjt** [comment]
> @christjt Don't we need this changes from `renderer.getSize()` to `renderer.getDrawingBufferSize()` in AxisViewTool?

@pramodcog We do not, since AxisViewTool is a UI element the physical size should be roughly equal on any device. If it is changed to `renderer.getDrawingBufferSize()`, it would be tiny on high resolutions devices (such as iPhones with retina).

**Comment by christjt** [comment]
> @christjt Does `potree` library take care of the pixel ratio?

Yes, it handles it internally.

**Pull request merged**